### DISCUSSION TITLE
Using QuestPDF with .NET MAUI

### DISCUSSION THREAD

**Discussion by DonkeyKongJr** [QuestPDF/QuestPDF#925] [Q&A] [answered]
Is there a chance to use QuestPDF with .NET MAUI to generate a PDF?

We have successfully tested QuestPDF with a PoC in an very simplistic .NET Console application and have recently began to port the PoC in our .NET MAUI app. However, after starting the app for the first time QuestPDF throws a System.Exception with the following content.
```

Your runtime is currently not supported by QuestPDF. Currently supported runtimes are: win-x86, win-x64, linux-x64, linux-arm64, linux-musl-x64, osx-x64, osx-arm64. 

Your current runtime is detected as 'other-x64'.'
```

I thought I have noticed that .NET Maui is supported. Did we miss something here?

The platform checks in the source code don't list android-x64 or ios-x64 as supported  tough.

**Comment by MarcinZiabek** [comment]
Thank you for sharing your feedback 😄 

Indeed, the MAUI platform has been supported up to the `2023.12.X` version. In the `2024.X.Y` cycle, we have switched from SkiaSharp to our own custom-built Skia layer. While this change has provided many opportunities and value, it has removed feedback for certain platforms.

Document generation happens primarily on servers, so we have prioritized those platforms. We are actively listening to the feedback. If there is significant usage on mobile platforms, we are happy to reevaluate the support. 

As an example, so far, the WASM platform is under active development/prototyping.

**Reply by troika-support** [reply]
I am running a Blazor ASPNET Core Hosted application with PDF generation capabilities. I was able to generate PDFs before within my app while it was running on .NET 7. On upgrading it to .NET 8 and updating the QuestPDF package it started showing me the following error when trying to run code that generates PDF:

'The QuestPDF library has encountered an issue while loading one of its dependencies.\n\nYour runtime is currently not supported by QuestPDF. Currently supported runtimes are: win-x86, win-x64, linux-x64, linux-arm64, linux-musl-x64, osx-x64, osx-arm64. \n\nYour current runtime is detected as 'browser-wasm'.'

What should I do to fix this for my solution?

**Reply by DonkeyKongJr** [reply]
That's really unfortunate because I quite like QuestPDF, and it's FluentAPI, however, our customer needs on device report generation, and it has to work when certain users are offline.

**Reply by Pnixys** [reply]
Hi, I'm also using .NET MAUI.
I feel quite sad that I can't use QuestPDF on MAUI. My team and I are facing a use case where the mobile app needs to generate a document, rather than the backend.

**Reply by Fidazahid** [reply]
We also need to generate reports in the .NET MAUI app

**Reply by NicolasKlaas** [reply]
Sad that this isn't possible yet.. (used to have this working on Xamarin with the 2023.x QuestPdf package). I am now also trying to use it in MAUI.. sadly have to move to Syncfusions solution now :'(

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Support for mobile platforms was dropped in version 2024.3.0 when QuestPDF transitioned to its own native Skia layer. QuestPDF does not currently support Xamarin, MAUI, Blazor WASM, or Uno, and there are no immediate plans to reintroduce support for these platforms. As PDF generation is primarily a backend-specific feature, we recommend generating documents on the server side. We may consider reintroducing support for mobile platforms depending on community demand, but it remains low in priority for the foreseeable future.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
